lib/scoreboard.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class Scoreboard
+  attr_reader :result
+
+  LOWEST_SCORE = [
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0],
+    [0, 0]
+  ].freeze
+
+  HIGHEST_SCORE = [
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10],
+    [10, 10, 10]
+  ].freeze
+  def initialize(frames)
+    @result = case frames
```

Comment:
Your class here is not really used as a class. It calculates and stores a result at initialisation, so it basically is just acting as a function.
Have you modelled what your program's input/output should be? In other words, how would you like the user to use your programme? What will they call, in what order?

Reply:
Have you tested with games that have strikes or spares in the tenth frame? I think your data model doesn't work in those circumstances. If there's a strike or a spare in the tenth frame, the player needs to roll more balls in order to calculate bonuses correctly, but these rolls do not count as a frame by themselves, only as bonuses.
In your data model, there's no good way to represent these additional rolls, as you are passing rolls that represent frames directly.

One option you could move to is to store an array of rolls instead of frames, letting the logic of your code decides which frames the rolls should belong to if any.